dags/iop/parser.py

```diff
@@ -100,3 +105,80 @@ def _get_extracted_arxiv_eprint_value(self, article):
             self.logger.error("The arXiv value is not valid.", dois=self.dois)
         except AttributeError:
             self.logger.error("No arXiv eprints found", dois=self.dois)
+
+    def _get_authors(self, article):
+        contrib_types = article.findall(
+            "front/article-meta/contrib-group/contrib[@contrib-type='author']"
+        )
+        authors = []
+        surname = ""
+        given_names = ""
+
+        for contrib_type in contrib_types:
+            try:
+                surname = contrib_type.find("name/surname").text
+            except AttributeError:
+                self.logger.error("Surname is not found in XML", dois=self.dois)
+            try:
+                given_names = contrib_type.find("name/given-names").text
+            except AttributeError:
+                self.logger.error("Given_names is not found in XML", dois=self.dois)
+
+            reffered_ids = contrib_type.findall("xref[@ref-type='aff']")
+            affiliations = [
+                self._get_affiliation_value(article, reffered_id)
+                for reffered_id in reffered_ids
+                if self._get_affiliation_value(article, reffered_id) is not None
+            ]
+
+            if "collaboration" not in given_names.lower():
+                author = clean_dict(
+                    {
+                        "surname": surname,
+                        "given_names": given_names,
+                        "affiliations": affiliations,
+                    }
+                )
+                authors.append(author)
+
+        if all(authors):
+            return authors
+
+    def _get_affiliation_value(self, article, reffered_id):
+        institution_and_country = {}
+        try:
+            id = reffered_id.get("rid")
+        except AttributeError:
+            self.logger.error("Referred id is not found")
+        institution = self._get_institution(article, id)
+        country = self._get_country(article, id)
+        try:
+            institution_and_country = {"country": country}
+            institution_and_country.update(
+                {"institution": ", ".join([institution, country])}
+            )
+        except TypeError:
```

Comment:
why is that, what types those values can have? And also why do you need to have a country in this dict and then the same information in an institution?

Reply:
1. If the country won't be in XML, it will be None. None + string (institution ) = TypeError
2. That's the schema: institution and country have to be in the dict. Also, we are using these values, if I am not mistaken, is important in country "share statistics"

Reply:
then maybe add to the logger `institution= institution, country= country`?